tests/test_main.py

```diff
@@ -60,14 +56,20 @@ def test_set_key(dotenv_file, before, key, value, expected, after):
 
 
 def test_set_key_permission_error(dotenv_file):
-    os.chmod(dotenv_file, 0o000)
+    result = sh.Command("whoami")()
+    result = str(result).strip()
 
+    os.chmod(dotenv_file, 0o000)
     with pytest.raises(Exception):
         dotenv.set_key(dotenv_file, "a", "b")
+        assert result != "root"
 
     os.chmod(dotenv_file, 0o600)
     with open(dotenv_file, "r") as fp:
-        assert fp.read() == ""
+        if result != "root":
+            assert fp.read() == ""
+        else:
+            assert fp.read() == "a=\"b\"\n"
```

Comment:
I'm a bit confused by what changed in this test. What's going here?

Reply:
The test was passing when run without root permissions, but it failed when I ran it as root. 

That is, I would run pytest on my windows subsystem for linux as a root user, and it would fail. I thought I'd modify the test so it would pass when run as root. I'd understand if it should be reverted.